httpql/src/main/java/com/hubspot/httpql/internal/CombinedFilterEntry.java

```diff
@@ -0,0 +1,107 @@
+package com.hubspot.httpql.internal;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.jooq.Condition;
+import org.jooq.Operator;
+import org.jooq.impl.DSL;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+import com.hubspot.httpql.FieldFactory;
+import com.hubspot.httpql.QuerySpec;
+
+public class CombinedFilterEntry<T extends QuerySpec> implements FilterEntryConditionCreator<T> {
+
+  private Operator operator;
+
+  private List<FilterEntryConditionCreator<T>> conditionCreators;
```

Comment:
Does order matter here? Would it be worth considering using a `Set` for this (would also ensure uniqueness, though not sure if that matters)?

Reply:
I don't really think order matters here but this would be a much deeper undertaking as all the tests for QueryParser and SelectBuilder are built on the assumption that order matters for their expected values. I originally wrote this as a set but gave up after fighting with the tests for a bit.